### PR TITLE
Migrations for indexes on batch tables

### LINKED ISSUE #5695: Further performance improvements for job executions (for oracle, other dbs might vary)
**Description:**
For Oracle we found out that the performance for job executions sqls can be further improved as they are not fast enought to deliver a proper response for the UI in time. (30 seconds then our gateway cuts off) Note that we already have 1743745 job executions.

**Release versions:**
SCDF 2.11.2

**Custom apps:**
N/A

**Steps to reproduce:**
Just go into the job executions page at the SCDF UI and navigate through the pages.

**Screenshots:**
image

**Additional context:**
The issue can be solved by ceating the following indexes.

```sql
create index SCDF.SYS_C0098765 on SCDF.BATCH_STEP_EXECUTION("JOB_EXECUTION_ID");
create index SCDF.SYS_C0087654 on SCDF.BOOT3_BATCH_STEP_EXECUTION("JOB_EXECUTION_ID");
create index SCDF.SYS_C0076543 on SCDF.BOOT3_TASK_TASK_BATCH("JOB_EXECUTION_ID");
create index SCDF.SYS_C0065432 on SCDF.BATCH_JOB_EXECUTION("START_TIME");
```

Also the oracle database suggests a sql_profile for the first page which can not be created via SQL statements.

After those changes the paged job executions can be delivered in ~900ms (for the initial page (1) and all following (2-N))

### PR DESCRIPTION
**Pull request by corneil**
Migrations for indexes on batch tables to improve aggregate view page queries.

Fixes #5695

Also added fix for `SimpleJobServiceMariadbTests` and `SimpleJobServicePostgresTests`

**DO NOT SQUASH** Use Rebase & Merge

### CODE REVIEW

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/db/migration/AbstractCreateBatchIndexesMigration.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2023 the original author or authors.
```
Comment (cppwfs) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/db/migration/AbstractCreateBatchIndexesMigration.java:2]: Update copyright date to 2024.

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/db/migration/AbstractCreateBatchIndexesMigration.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import java.util.List;
+
+import org.springframework.cloud.dataflow.common.flyway.AbstractMigration;
+import org.springframework.cloud.dataflow.common.flyway.SqlCommand;
+
+/**
+ * Provide indexes to improve aggregate view performance
+ * @author Corneil du Plessis
+ */
+public abstract class AbstractCreateBatchIndexesMigration extends AbstractMigration {
```
Comment (cppwfs) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/db/migration/AbstractCreateBatchIndexesMigration.java:28]: Are these changes related to issue: https://github.com/spring-cloud/spring-cloud-dataflow/issues/5695 ?

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/db/migration/AbstractCreateBatchIndexesMigration.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	protected static final String CREATE_BATCH_STEP_EXECUTION_JOB_EXECUTION_ID_INDEX = "create index BATCH_STEP_EXECUTION_JOB_EXECUTION_ID_IX on BATCH_STEP_EXECUTION(JOB_EXECUTION_ID)";
```
Comment (cppwfs) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/db/migration/AbstractCreateBatchIndexesMigration.java:29]: Just a nit, can we put the SQL on another line so we don't have to pan to view it.  
Thanks!

**Review by cppwfs — COMMENTED** [review]
Can you provide the reason why we need the optimizations?

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/db/migration/AbstractCreateBatchIndexesMigration.java (hunk 2 above)
Comment (cppwfs) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/db/migration/AbstractCreateBatchIndexesMigration.java:28]: Ahh.  Sorry I see it in your comment.

**Comment by corneil** [comment]
> Can you provide the reason why we need the optimizations?

Customer experienced queries so long that their ingress requests timed out at 120 or 180s.
After adding indexes the queries came in below 1s.

**Review by onobc — APPROVED** [review]
Thanks for the fixes @corneil - LGTM !

**Comment by klopfdreh** [comment]
That was fast. Thanks a lot! 👍

**Review by klopfdreh — APPROVED** [review]
Looks good!

**Pull request merged**